R/plotBoxplot.R

```diff
@@ -359,6 +363,27 @@ setMethod("plotBoxplot", signature = c(object = "SummarizedExperiment"),
         stop("'facet.by' must not match with 'x', 'group.by' or 'fill.by'.",
             call. = FALSE)
     }
+
+    # p.value defines p-values to be plotted. It should be in specific format.
+    # It should have p-values and groups along with facetting variable
+    # if it was specified.
+    if( is.null(x) && !is.null(p.value) ){
+        stop("If 'x' is not specified, 'p.value' must be NULL.", call. = FALSE)
+    }
+    if( !(is.null(p.value) || is.data.frame(p.value)) ){
+        stop("'p.value' must be NULL or data.frame.", call. = FALSE)
+    }
+    cols <- c("group1", "group2")
+    p_cols <- c("p", "p.adj", "p.adj.signif")
+    if( is.data.frame(p.value) && (
+            !all(cols %in% colnames(p.value)) ||
+            sum(p_cols %in% colnames(p.value)) == 0L)
+        ){
+        stop("'p.value' must have columns '", paste0(cols, collapse = "', '"),
+            "' and one of the following columns: '",
+            paste0(p_cols, collapse = "', '"), "'", call. = FALSE)
```

Comment:
What if `facet.by`, `group.by` and `fill.by` are specified, but p values do not have the same grouping?

For instance, if p values includes only one p-value. And not multiple groups that are visualized.

Probably does not work correctly

Reply:
The input check should be improved, it is hard for user to know what to do (what is the correct format)